apps/op-func/src/infra/storage/access-log-register.ts

```diff
@@ -0,0 +1,61 @@
+import { BlockBlobUploadResponse, ContainerClient } from "@azure/storage-blob";
+import {
+  AuditEvent,
+  auditEventSchema,
+} from "io-fims-common/domain/audit-event";
+import * as assert from "node:assert/strict";
+
+// A helper function used to read a Node.js readable stream into a String
+export async function getStreamIntoString(
+  readableStream: NodeJS.ReadableStream,
+) {
+  const chunks = [];
+
+  for await (const chunk of readableStream) {
+    chunks.push(Buffer.from(chunk));
+  }
+
+  return Buffer.concat(chunks).toString("utf-8");
+}
+
+export class AccessLogRegister {
+  #container: ContainerClient;
+
+  constructor(client: ContainerClient) {
+    this.#container = client;
+  }
+
+  async get(name: string): Promise<AuditEvent> {
+    try {
+      const blobClient = this.#container.getBlobClient(name);
```

Comment:
I would move this out of the try as we did for the upload just for consistency

Reply:
Actually I think it's better to put everything inside the try (also in the upload) in order to catch all the possible errors